engine/src/lib.rs

```diff
@@ -481,6 +555,30 @@ mod contract {
         io.return_output(block_hash.as_bytes())
     }
 
+    #[no_mangle]
+    pub extern "C" fn get_previous_block_hashchain() {
+        let mut io = Runtime;
+        let block_height = io.block_height();
+        let mut blockchain_hashchain = hashchain::get_state(&io).sdk_unwrap();
+
+        if block_height > blockchain_hashchain.get_current_block_height() {
+            blockchain_hashchain
+                .move_to_block(block_height)
+                .sdk_unwrap();
+        }
+
+        let previous_block_hashchain = blockchain_hashchain.get_previous_block_hashchain();
+        io.return_output(&previous_block_hashchain);
+    }
+
+    #[no_mangle]
+    pub extern "C" fn get_genesis_block_hashchain() {
```

Comment:
Similar comment here about returning a tuple `<block_height, block_hashchain>` so it is clear where the genesis hashchain started.

Reply:
Genesis will start at 0 (this is the standard for Ethereum-like chains), so I do not think the block height is needed here.